heartbeat/README.mariadb-semi-sync.md

```diff
@@ -0,0 +1,149 @@
+Setting up the MariaDB Semi Sync resource agent
+===============================================
+
+Before embarking on this quest one should read the MariaDB pages on replication
+and global transaction IDs, GTID. This will greatly help in understanding what
+is going on and why.
+
+Part 1: MariaDB Setup
+---------------------
+
+It is best to initialize your MariaDB and do a failover before trying to use
+Pacemaker to manage MariaDB. This will both verify the MariaDB configuration
+and help you understand what is going on.
+
+###Configuration Files
+
+In your MariaDB config file for the server on node 1, place the following
+entry (replacing my_database and other names as needed):
+```
+[mariadb]
+log-bin
+server_id=1
+log-basename=master
+binlog_do_db=my_database
+```
+
+Then for each other node create the same entry, but increment the server_id.
+
+###Replication User
+
+Now create the replication user:
+```
+GRANT ALL PRIVILEGES ON *.* TO 'slave_user'@'%' IDENTIFIED BY 'password';
+GRANT ALL PRIVILEGES ON *.* TO 'slave_user'@'localhost' IDENTIFIED BY 'password';
+
+FLUSH PRIVILEGES;
+```
```

Comment:
I'd explicitly mention that users should replace 'password' with their password, you'd be surprised how many people copy-and-paste without looking :-)

There's no need to flush privileges here -- that's a bit of voodoo handed down over the years that's almost never necessary. It's only needed when changing privileges for an account that is currently logged in, and you want the existing login session to get the new privileges.